picasso-layoutlib-compat/src/main/java/com/squareup/picasso3/LayoutLibThreadFactory.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.picasso3
+
+import android.annotation.SuppressLint
+import android.os.Handler_Delegate
+import com.android.layoutlib.bridge.impl.RenderAction
+import java.util.concurrent.ThreadFactory
+
+/**
+ * LayoutLib requires assistance bridging the gap to work with ThreadFactory.
+ * This workaround ensures the Handler_Delegate executes all threads that are created
+ * by Picasso allowing RequestHandler's the chance to fulfill a request synchronously.
+ */
+@SuppressLint("SyntheticAccessor")
+class LayoutLibThreadFactory : ThreadFactory {
+
+  init {
+    if (layoutLibEnvironment) {
+      // Handler_Delegate sendMessageAtTime drops messages that
+      // don't contain a callback. So we will continue to check
+      // if the message has a callback and add it to the
+      // handlerMessageQueue like normal else check if it is
+      // Picasso's Action trying to be sent, at which point
+      // we want to forward that message to continue processing
+      Handler_Delegate.setCallback { handler, message, uptimeMillis ->
+        if (message.callback != null) {
+          val context = RenderAction.getCurrentContext() ?: return@setCallback
+          context.sessionInteractiveData.handlerMessageQueue.add(
+            handler,
+            uptimeMillis,
+            message.callback
+          )
+        } else if (ACTION.isInstance(message.obj)) {
+          handler.handleMessage(message)
+        }
+      }
+    }
+  }
+
+  override fun newThread(runnable: Runnable) =
+    object : Thread(runnable) {
+      override fun run() {
+        // This is Picasso's thread, we always want to handle messages posted here
+        if (layoutLibEnvironment)
+          Handler_Delegate.setCallback { handler, message, _ -> handler.handleMessage(message) }
+
+        super.run()
+      }
+    }
+
+  companion object {
```

Comment:
Super nit: is there a reason to put `private` on the children instead of on the `companion object`?

Reply:
agreed, this should be private